### ISSUE TITLE
Zombie process

### ISSUE THREAD

**Issue by Ty3uK** [evanw/esbuild#1566]
Hi!

In my project I'm using esbuild inside worker_threads. In watch mode worker starts on external file change (I'm using chokidar).

On macOS 11.5.2 I noticed that esbuild process becomes a zombie after successfull build.

![CleanShot 2021-09-01 at 00 49 45@2x](https://user-images.githubusercontent.com/2384454/131580569-55ce6453-5144-48f9-a671-e591fc24c228.png)

Also I found some similar issue - #643

**Comment by evanw** [comment]
From the graph, it looks like the parent process is still running? To me, a zombie process means one where the child continues without exiting even after the parent exits. Using the JS esbuild API starts a long-lived child process for the rest of the session that is reused for future JS API calls. So what you are showing appears to be expected behavior.

**Comment by Ty3uK** [comment]
Yep, parent process is running, but inside worker I called `process.exit()`.

Calling `child.kill()` from `esbuild/lib/main.js` has no effect too.

**Comment by Ty3uK** [comment]
> To me, a zombie process means one where the child continues without exiting even after the parent exits

I got into a situation where I had 6 esbuild processes and `htop` marks all of them as `Z`

**Comment by evanw** [comment]
That does sounds like a zombie process then. Can you provide instructions for how to replicate your situation (or provide some code to demonstrate the issue)?

**Comment by Ty3uK** [comment]
Yep, I will make a repro code for you

**Comment by Ty3uK** [comment]
Sorry for delay.

https://github.com/Ty3uK/esbuild-zombie-reproduce

**Comment by Ty3uK** [comment]
Thank you for so detailed answer!
